test/unit/lib/plugins/aws/package/compile/events/alb/index.test.js

```diff
@@ -219,4 +217,75 @@ describe('lib/plugins/aws/package/compile/eents/alb/index.test.js', () => {
         .and.have.property('code', 'ALB_NO_CONDITIONS');
     });
   });
+
+  describe('should support `functions[].events[].alb.targetGroupName` property', () => {
+    it('should use it if defined', async () => {
+      const result = await runServerless({
+        fixture: 'function',
+        cliArgs: ['package'],
+        configExt: {
+          functions: {
+            fnTargetGroupName: {
+              handler: 'index.handler',
+              events: [
+                {
+                  alb: {
+                    ...validBaseEventConfig,
+                    priority: 1,
+                    targetGroupName: 'custom-targetgroup-name',
+                  },
+                },
+              ],
+            },
+          },
+        },
+      });
+
+      const albListenerRuleLogicalId = naming.getAlbTargetGroupLogicalId(
+        'fnTargetGroupName',
+        albId,
+        false
+      );
+
+      expect(result.cfTemplate.Resources[albListenerRuleLogicalId].Properties.Name).to.equal(
+        'custom-targetgroup-name'
+      );
+    });
+
+    it('should reject if `provider.alb.targetGroupPrefix` is also specified', async () => {
+      const runServerlessAction = () =>
+        runServerless({
+          fixture: 'function',
+          cliArgs: ['package'],
```

Comment:
Rebase against master and put `command: 'package'` instead (we've just upgraded to new version of `runServerless` -> https://github.com/serverless/serverless/pull/9281)